### PR TITLE
Fix how we handle hidden and deprecated packages

### PR DESCRIPTION
**Pull request by annafil**
This PR fixes a number of issues with how we've been treating hidden and deprecated packages: 

1. The code to prevent packages from crawling hidden pages wasn't working at all -- it was hard wired not to hide the latest version of any package, behavior we don't actually want. The changes in this PR enable both individual packages to be hidden from crawlers, as well as entire org pages (e.g. fishtown-analytics). 

Screen Shot 2021-08-20 at 1 36 12 PM

2. The existing structure did not have a means of redirecting users to new versions of packages, if older ones were deprecated. This now exits and lives in the package configuration file alongside logo assets. The fields are optional. 

    **NB:** This only works on the web version of the hub. I did not make any modifications to the API content being served, because this is in the critical path of package installation and I prefer to wait for someone from the Core team to 🍐 on this one. **Therefore these new fields will NOT show up when Core queries the static API**. 

Screen Shot 2021-08-20 at 1 36 41 PM

3. Hiding by organization and package name was inconsistent. E.g. packages that should be hidden from the organization page because they have been archived, were not. Packages are now correctly removed from the author page. 

Screen Shot 2021-08-20 at 1 17 55 PM

4. There is now a way to specify redirects both by package namespace as well as name (e.g. making sure both fishtown-analytics/dbt-utils and fishtown-analytics/dbt_utils redirect appropriately to dbt-labs/dbt_utils). 
Screen Shot 2021-08-20 at 12 44 35 PM

5. It is now possible to blocklist packages and completely remove them from the hub listing (relevant for the initial list of archived packages like dbt-labs/heap) AND set up an archive notice OR set up a redirect notice. This should be more intuitive for users rather than a "File not found" message they'd get previously. 

Screen Shot 2021-08-20 at 12 44 07 PM

6. Warnings are shown at the very top of the site in new web colors, to be consistent with www. I also took the liberty of updating button colors. 

7. Some ruby configuration has been updated to enable the above, and to use Ruby 2.7 -- middleman no longer starts without this.

### CODE REVIEW

**Comment by annafil** [comment]
This PR is a short-term fix for issue #810

**Comment by joellabes** [comment]
https://github.com/dbt-labs/dbt/pull/3759 is not strictly related to this and #810, but they're cousins. Specifically, I'd be interested in seeing `dbt deps` being able to warn about the `fishtown-analytics` ➡️ `dbt-labs` namespace migration but I don't think it make sense to warn about archived packages (e.g. heap). 

Ultimately I think this becomes another issue on the Core repo, but wanted to throw it into the mix while it's in the Zeitgeist

**Comment by sungchun12** [comment]
> [dbt-labs/dbt#3759](https://github.com/dbt-labs/dbt/pull/3759) is not strictly related to this and #810, but they're cousins. Specifically, I'd be interested in seeing `dbt deps` being able to warn about the `fishtown-analytics` ➡️ `dbt-labs` namespace migration but I don't think it make sense to warn about archived packages (e.g. heap).
> 
> Ultimately I think this becomes another issue on the Core repo, but wanted to throw it into the mix while it's in the Zeitgeist

I agree that this requires a NEW issue given a new process needs to be created overall for renaming and deprecating packages outlined in #810.

File: data/packages/fishtown-analytics/adwords/index.json [hunk 1]
```diff
@@ -5,5 +5,6 @@
     "latest": "0.2.12",
     "assets": {
         "logo": "logos/placeholder.svg"
-    }
+    },
+    "redirectnamespace": "dbt-labs"
```
Comment (drewbanin) [inline data/packages/fishtown-analytics/adwords/index.json:9]: i think these `index.json` files will be overwritten every time the [hubcap script](https://github.com/dbt-labs/hubcap/blob/master/hubcap.py#L258-L261) runs. This makes a ton of sense as a backfill, but should we also update the hubcap script to either:
1. Avoid overwriting this index file entirely (we'd still need to update the `latest` key)
2. Set the `redirectnamespace` and `redirectname` for these packages

**Comment by drewbanin** [comment]
One comment here around how this interplays with the hubcap script, but otherwise looking great!

File: data/packages/fishtown-analytics/adwords/index.json (hunk 1 above)
Comment (annafil) [reply data/packages/fishtown-analytics/adwords/index.json:9]: Good point @drewbanin! I'm not sure that hubcap will be overwriting these old folders I've been updating -- that's kind of the problem 😅  Hubcap only updates the latest location of the package, and not these older folders which are the ones that have redirects. 

That being said, I'm sure we'll need to spend more time thinking about better naming conventions and where this redirect information should eventually live to be picked up by Core's dbt deps. I'm 💯 on doing iterations on this format after we apply the short term fix.

Comment (drewbanin) [reply data/packages/fishtown-analytics/adwords/index.json:9]: 🤦 🤦 🤦 yes..... you're so right..... thanks for helping me get there :)

**Review by jtcohen6 — APPROVED** [review]
This looks fantastic!

> Therefore these new fields will NOT show up when Core queries the static API.

Let's spin up a new PR that adds these fields to the static API, and test `dbt deps` pointed to the staging deploy. If that works, it will be a first step toward adding some better logic to Core that can handle redirects / aliases, as we discussed.

**Pull request merged**